src/main/kotlin/com/github/imflog/schema/registry/tasks/download/DownloadTaskAction.kt

```diff
@@ -33,6 +34,29 @@ class DownloadTaskAction(
         return errorCount
     }
 
+    private fun expandSubjectPatterns() : List<DownloadSubject> {
+        val allSubjects = client.allSubjects
+        return subjects.flatMap { downloadSubject ->
+            if (downloadSubject.regex) {
+                val regex = parseSubjectRegex(downloadSubject.subject)
+                allSubjects.filter { subject -> regex?.matches(subject) ?: false }
+                    .map { subject -> DownloadSubject(subject, downloadSubject.file, downloadSubject.version) }
+                    .toList()
+            } else {
+                listOf(downloadSubject)
+            }
+        }
+    }
+
+    private fun parseSubjectRegex(regex: String): Regex? {
+        return try {
+            Regex(regex)
+        } catch (exception: PatternSyntaxException) {
+            logger.error("Unable to compile subject pattern of ${regex}", exception)
```

Comment:
nit: the {} are not mandatory here 
Also, can you add the fact that we are skipping ? Something like:
`Unable to compile subject pattern of $regex, skipping it` ?

Reply:
Can you tell that I haven't done much kotlin :-D

Changed the error message as suggested.